set_id.go

```diff
@@ -16,16 +17,20 @@ func (s SetID) String() string {
 
 // newSetID calculates a deterministic hash over a set of htlc circuit
 // keys.
-func newSetID(keys []types.CircuitKey) SetID {
-	sortedKeys := make([]types.CircuitKey, len(keys))
+func newSetID(keys []types.HtlcKey) SetID {
+	sortedKeys := make([]types.HtlcKey, len(keys))
 	copy(sortedKeys, keys)
 
 	sort.Slice(sortedKeys, func(a, b int) bool {
-		if sortedKeys[a].ChanID == sortedKeys[b].ChanID {
-			return sortedKeys[a].HtlcID < sortedKeys[b].HtlcID
+		if sortedKeys[a].Node != sortedKeys[b].Node {
```

Comment:
Fixing the hash collision!

Reply:
What happened if `len(keys)==1`?

we could still have the same set_id if we make a payment A-->B or B-->A ?

Reply:
I mean, in case of self-payment:

NewSetID(nodeA,htlcID,channelID) == NewSetID(nodeB,htlcID,channelID)

Reply:
Oops, forgot to add the node key 😬

Reply:
Also changed it to use a writer, to avoid the indexing logic.